packages/nimble-components/src/table-column/base/index.ts

```diff
@@ -45,5 +48,27 @@ export abstract class TableColumn<
             );
         }
         this.columnInternals = new ColumnInternals(options);
+        this.columnInternals.currentSortDirection = this.sortDirection;
+        this.columnInternals.currentSortIndex = this.sortIndex;
+    }
+
+    protected sortDirectionChanged(): void {
+        this.columnInternals.currentSortDirection = this.sortDirection;
```

Comment:
Does there need to be a check within `sortDirectionChanged` and `sortIndexChanged` ensuring that sorting is enabled? The idea is that the `currentSort*` values are always `undefined` when sorting is disabled, right?

Perhaps there should be a test that:
- disables sorting on the column
- updates the `sort-direction` and `sort-index` attributes on that column
- verifies `columnInternals.currentSortDirection` and `columnInternals.currentSortIndex` are undefined
- re-enables sorting on the column and verifies `columnInternals.currentSortDirection` and `columnInternals.currentSortIndex` are set to the values configured while sorting was disabled

Reply:
You're right.  
Updated the code so currentSortIndex/Direction won't get updated from sortIndex/sortDirection while sortingDisabled is true.  
Added that test in table-column.spec.ts.